### ISSUE TITLE
Queries containing `REGEXP`/`IREGEXP` can not be parsed using the `hive` dialect

### ISSUE THREAD

**Issue by vsevolodbazhan** [sqlfluff/sqlfluff#4831]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

## Context

I am using SQLFluff with the `hive` dialect to lint [Impala](https://impala.apache.org/) queries. Hive and Impala have highly compatible dialects so there is no problem for the most part except `REGEXP`/`IREGEXP` operator.

## Problem

SQLFluff fails to parse Impala/Hive queries that contain `REGEXP`/`IREGEXP`.

## Notes
- `RLIKE` which is effectively a synonym to `REGEXP` works correctly.
- The issue #2776 seem to report a similar problem but for Snowflake dialect.

### Expected Behaviour

Such queries to be parsed correctly.

### Observed Behaviour

```
Found 1 issues while parsing string.
Line 4, Position 14: Found unparsable section: " REGEXP '(/click)'"
Found 1 issues while parsing string.
Line 4, Position 13: Found unparsable section: "REGEXP '(/click)'"
Found 1 issues while parsing string.
Line 4, Position 14: Found unparsable section: " IREGEXP '(/click)'"
Found 1 issues while parsing string.
Line 4, Position 21: Found unparsable section: "'(/click)'"
```

### How to reproduce

```python
import sqlfluff

queries = (
    """
    SELECT *
    FROM example
    WHERE url REGEXP '(/click)';
    """,
    """
    SELECT
        *,
        url REGEXP '(/click)'
    FROM example;
    """,
    """
    SELECT *
    FROM example
    WHERE url IREGEXP '(/click)';
    """,
    """
    SELECT
        *,
        url IREGEXP '(/click)'
    FROM example
    """,
    """
    SELECT *
    FROM example
    WHERE url RLIKE '(/click)'
    """,
    """
    SELECT
        *,
        url RLIKE '(/click)'
    FROM example;
    """
)

for query in queries:
    try:
        sqlfluff.parse(query, dialect="hive")
    except Exception as error:
        print(error)
```

### Dialect

Hive

### Version

❯ python --version
Python 3.11.2

❯ sqlfluff --version
sqlfluff, version 2.0.7

### Configuration

No custom configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
